services/harmony/app/frontends/ogc-edr/index.ts

```diff
@@ -0,0 +1,91 @@
+import * as yaml from 'js-yaml';
+import getLandingPage from '../ogc-coverages/get-landing-page';
+import getRequirementsClasses from '../ogc-coverages/get-requirements-classes';
+
+import getDataForCube from './get-data-for-cube';
+import postDataForCube from './post-data-for-cube';
+
+import HarmonyRequest from '../../models/harmony-request';
+
+interface OgcSchemaHttpMethod {
+  parameters: {
+    $ref: string
+  }[]
+}
+
+interface OgcSchemaEdr {
+  paths : {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    '/collections/{collectionId}/cube': {
+      get: OgcSchemaHttpMethod,
+      post: OgcSchemaHttpMethod
+    }
+  }
+}
+
+export const version = '1.0.0';
+const openApiRoot = path.join(__dirname, '..', '..', 'schemas', 'ogc-api-edr', version);
+const openApiPath = path.join(openApiRoot, `ogc-api-edr-v${version}.yml`);
+export const openApiContent = fs.readFileSync(openApiPath, 'utf-8');
+const ogcSchemaEdr = yaml.load(openApiContent, { schema: yaml.DEFAULT_SCHEMA }) as OgcSchemaEdr;
+export const edrGetParams = ogcSchemaEdr
+  .paths['/collections/{collectionId}/cube'].get.parameters
+  .map(param => param.$ref.split('/').pop());
+export const edrPostParams = ['shapefile'].concat(ogcSchemaEdr
+  .paths['/collections/{collectionId}/cube'].post.parameters
+  .map(param => param.$ref.split('/').pop()));
+
+/**
+ * Express handler that returns a 501 error and "not yet implemented" message to the client
+ *
+ * @param req - The request sent by the client
+ * @param res - The response to send to the client
+ */
+function TODO(req: HarmonyRequest, res: Response): void {
+  res.status(501);
+  res.json('Not yet implemented');
+}
+
+/**
+ * Express handler that returns the OpenAPI spec for a collection
+ *
+ * @param req - The request sent by the client
+ * @param res - The response to send to the client
+ */
+function getSpecification(req: HarmonyRequest, res: Response): void {
+  // Defined inline because the index file deals with the YAML spec.
+  res.append('Content-type', 'text/openapi+yaml;version=3.0');
+  res.send(openApiContent);
+}
+
+/**
+ * Sets up the express application with the OpenAPI routes for OGC API - EDR
+ *
+ * @param app - The express application
+ */
+export function addOpenApiRoutes(app: Router): void {
+  initialize({
+    app: app as Application,
+    apiDoc: openApiContent,
+    validateApiDoc: true,
+    /* Note: the default way to expose an OpenAPI endpoint is to have express handle paths
+     * based on a supplied directory structure. Instead we are using the operations property
+     * because we want to include the paths within the OpenAPI specification itself. */
+    operations: {
+      getLandingPage,
+      getRequirementsClasses,
+      getSpecification,
+      describeCollections: TODO,
+      describeCollection: TODO,
+      getCollectionInstances: TODO,
+      getDataForPoint: TODO,
+      postDataForPoint: TODO,
+      getDataForCube,
```

Comment:
Should we add TODOs for the other paths now as well, e.g. `getDataForArea`?

Reply:
If I remember correctly, we have a separate ticket for writing the EDR schema. I think this is part of that ticket.